Detectors/ITSMFT/ITS/simulation/src/Detector.cxx

```diff
@@ -145,7 +145,10 @@ Detector::Detector(Bool_t active, TString name)
     mDescriptorIB.reset(new DescriptorInnerBarrelITS2(3));
   } else if (name == "IT3") {
 #ifdef ENABLE_UPGRADES
-    mDescriptorIB.reset(new DescriptorInnerBarrelITS3(DescriptorInnerBarrelITS3::ThreeLayersNoDeadZones));
+    mDescriptorIB.reset(new DescriptorInnerBarrelITS3());
+    if (its3Version != "") {
+      dynamic_cast<DescriptorInnerBarrelITS3*>(mDescriptorIB.get())->setVersion(its3Version.Data());
```

Comment:
Out of curiosity, why a `dynamic_cast` here?

Reply:
There is no specific reason indeed, so I changed it (and the others) and tested it that it works properly using simple casts as well.